python_modules/dagster/dagster/_config/pythonic_config/config.py

```diff
@@ -85,7 +97,11 @@ def __setattr__(self, name: str, value: Any):
             return super().__setattr__(name, value)
         except (TypeError, ValueError) as e:
             clsname = self.__class__.__name__
-            if "is immutable and does not support item assignment" in str(e):
+            if "Instance is frozen" in str(  # Pydantic 2.x error
+                e
+            ) or "is immutable and does not support item assignment" in str(  # Pydantic 1.x error
+                e
+            ):
```

Comment:
is there any less fragile way of doing this?

Reply:
Still relevant comment

Reply:
Sadly no, not that I can tell. These values aren't set or exported as constants anywhere, just passed directly to the error constructor.